packages/astro/src/vite-plugin-astro-server/index.ts

```diff
@@ -423,8 +423,14 @@ async function handleRoute(
 			if (computedMimeType) {
 				contentType = computedMimeType;
 			}
-			res.writeHead(200, { 'Content-Type': `${contentType};charset=utf-8` });
-			res.end(result.body);
+      const response = new Response(result.body, {
+        status: 200,
+        headers: {
+          'Content-Type': `${contentType};charset=utf-8`,
+        },
+      });
+      attachToResponse(response, result.cookies);
+      await writeWebResponse(res, response);
```

Comment:
I converted the direct writing to the response to a call to `writeWebResponse` in order to streamline the response handling. Also added a call to `attachToResponse` in order to connect the `cookie` to the response.